src/CollectionV1.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+import "openzeppelin-contracts-upgradeable/contracts/utils/structs/EnumerableSetUpgradeable.sol";
+import "./IVisitable.sol";
+
+
+abstract contract CollectionV1 is Initializable {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+
+    EnumerableSetUpgradeable.AddressSet private _museums;
+    string public name;
+    uint8 public decimals; // Used when the origami is integer based
+    uint256 public version;
+    uint256 public counter;
+
+    struct OrigamiItem {
+        int256 data; // TODO: Default implementation is a uint256, change this and deploy for different data types, also requires changing other methods
```

Comment:
Did you think any about how we might make this dynamic based on the oracle type.

Reply:
It can be abi encoded then the client can do `abi.decode`, though it does not much thinking so I left it out of this PR